pkg/persistence/rethinkdb/store.go

```diff
@@ -0,0 +1,99 @@
+package rethinkdb
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/etc"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/harbor"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/job"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/persistence"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/xerrors"
+	r "gopkg.in/rethinkdb/rethinkdb-go.v6"
+)
+
+type rethinkdbScanJob struct {
+	LastUpdatedAt time.Time `json:"last_updated_at"`
+	job.ScanJob
+}
+
+type store struct {
+	conn r.QueryExecutor
+	cfg  etc.Rethink
+}
+
+func NewStore(conn r.QueryExecutor, cfg etc.Rethink) persistence.Store {
+	return &store{
+		conn: conn,
+		cfg:  cfg,
+	}
+}
+
+func (s *store) Create(scanJob job.ScanJob) error {
+	logrus.WithFields(logrus.Fields{
+		"scan_job_id":     scanJob.ID,
+		"scan_job_status": scanJob.Status.String(),
+		"rethinkdb_table": s.cfg.ScansTable,
+	}).Debug("Saving scan job")
+
+	rsj := rethinkdbScanJob{
+		LastUpdatedAt: time.Now().UTC(),
+		ScanJob:    scanJob,
+	}
+
+	res, err := r.Table(s.cfg.ScansTable).Insert(rsj).RunWrite(s.conn)
+	if err != nil || res.Inserted != 1 {
+		return xerrors.Errorf("inserting scan job failed: %w", err)
+	}
+
+	return nil
+}
+
+func (s *store) Get(scanJobID string) (*job.ScanJob, error) {
+	scanJob := new(job.ScanJob)
+	if err := r.Table(s.cfg.ScansTable).Get(scanJobID).ReadOne(scanJob, s.conn); err != nil {
```

Comment:
scanJobID looks like it should be primary index.  I believe since it's only annotated as json: id, the field name for this field won't be `id`.

Reply:
Same as above. Also, I've tested that everything works and there are integration tests for the store that are also passing.